src/processor.js

```diff
@@ -51,9 +51,11 @@ function traverse(node, callbacks) {
         callbacks["*"]();
     }
 
-    if (typeof node.children !== "undefined") {
-        for (let i = 0; i < node.children.length; i++) {
-            traverse(node.children[i], callbacks);
+    const parent = /** @type {ParentNode} */ (node);
+
+    if (typeof parent.children !== "undefined") {
+        for (let i = 0; i < parent.children.length; i++) {
+            traverse(parent.children[i], callbacks);
```

Comment:
An [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) like this is one common way of _"if it's a particular sub-type, run some corresponding logic"_ in TypeScript-land. However, a more idiomatic way would be to use [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html), such as by writing a [type predicate function](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates):

```suggestion
    if (isParentNode(node)) {
        for (let i = 0; i < node.children.length; i++) {
            traverse(node.children[i], callbacks);
```

```ts
/**
 * Determines whether the node is a parent (has children).
 * @param {Node} node A Markdown AST node.
 * @returns {node is ParentNode} Whether the node is a parent.
 */
function isParentNode(node) {
    return "children" in node;
}
```

Type predicates are roughly a more strict, subdued form of type assertions. They more clearly indicate _"if the value has some characteristics, then we can narrow its type to something useful"_.

Reply:
Ah nice! I like it.